pkgs/top-level/make-tarball.nix

```
@@ -44,11 +44,7 @@ pkgs.releaseTools.sourceTarball {
   checkPhase = ''
     echo "generating packages.json"
 
-    (
-      echo -n '{"version":2,"packages":'
-      NIX_STATE_DIR=$TMPDIR NIX_PATH= nix-env -f $src -qa --meta --json --show-trace --arg config 'import ${./packages-config.nix}'
-      echo -n '}'
-    ) | sed "s|$src/||g" | jq -c > packages.json
+    NIX_STATE_DIR=$TMPDIR NIX_PATH= nix-instantiate --eval --raw --expr "(import ${./packages-info.nix} { lib = import $src/lib; }) (import $src { config = import ${./packages-config.nix};})" | sed "s|$src/||g" | jq -c > packages.json
```

I think we should just do something like the following instead of interpolating `$src` in here and so on:


```suggestion
    NIX_STATE_DIR=$TMPDIR NIX_PATH= nix-instantiate --eval --raw ${./packages-info.nix} | sed "s|$src/||g" | jq -c > packages.json
```

(untested)

---

Needed to import `packages-config.nix` from `$src` as otherwise it can't see nixpkgs.